pkg/kubelet/kubelet_pods.go

```diff
@@ -465,12 +467,16 @@ func nodeHostsFileContent(hostsFilePath string, hostAliases []v1.HostAlias) ([]b
 func managedHostsFileContent(hostIPs []string, hostName, hostDomainName string, hostAliases []v1.HostAlias) []byte {
```

Comment:
@danwinship we are passing the hostIPs as a parameter, that are the PodIPs ... so I think that we should use it to make the decision to populate the localhost ipv6 addresses
```
func isSupportIPv6(hostIPs []string) bool {
    for _, ip := range hostIPs {
         if netutils.IsIPv6String(ip) {
                  return true
             }
     }
     return false
}
```

Reply:
This sounds tempting and simple enough :)

Reply:
`lo` will likely have `::1` even in single-stack IPv4 clusters, so if we want `/etc/hosts` to be "accurate", it can't be based on hostIPs.

Reply:
those `hostIPs` are the PodIPs

Reply:
@danwinship I think the podIPs parameters is the right way to decide to add ipv6 entries on the hosts file, as an IPv4 only pod will not need those addresses 

-- func (kl *Kubelet) GenerateRunContainerOptions(ctx context.Context, pod *v1.Pod, container *v1.Container, podIP string, podIPs []string)
--- makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, hostDomain string, podIPs ...
----- makeHostsMount(podDir string, podIPs []string,
-------ensureHostsFile(fileName string, hostIPs []string,
--------- managedHostsFileContent(hostIPs []string, hostName, ...

Reply:
> an IPv4 only pod will not need those addresses

_all_ IPv4 pods won't need those addresses?

Given that it has always been present in the past, I'm reluctant to remove it. If the address `::1` exists in a pod, it's more consistent to have `localhost` point to it.

Reply:
fair